# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../uclibc/gcc43.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Index: uClibc/include/sys/cdefs.h
===================================================================
--- uClibc/include/sys/cdefs.h	(revision 20731)
+++ uClibc/include/sys/cdefs.h	(revision 20732)
@@ -1,4 +1,5 @@
-/* Copyright (C) 1992-2001, 2002, 2004, 2005 Free Software Foundation, Inc.
+/* Copyright (C) 1992-2001, 2002, 2004, 2005, 2006, 2007
+   Free Software Foundation, Inc.
    This file is part of the GNU C Library.
 
    The GNU C Library is free software; you can redistribute it and/or
@@ -130,8 +131,18 @@
 /* Fortify support.  */
 #define __bos(ptr) __builtin_object_size (ptr, __USE_FORTIFY_LEVEL > 1)
 #define __bos0(ptr) __builtin_object_size (ptr, 0)
-#define __warndecl(name, msg) extern void name (void)
 
+#if __GNUC_PREREQ (4,3)
+# define __warndecl(name, msg) \
+  extern void name (void) __attribute__((__warning__ (msg)))
+# define __warnattr(msg) __attribute__((__warning__ (msg)))
+# define __errordecl(name, msg) \
+  extern void name (void) __attribute__((__error__ (msg)))
+#else
+# define __warndecl(name, msg) extern void name (void)
+# define __warnattr(msg)
+# define __errordecl(name, msg) extern void name (void)
+#endif
 
 /* Support for flexible arrays.  */
 #if __GNUC_PREREQ (2,97)
@@ -286,6 +297,31 @@
 # define __always_inline __inline
 #endif
 
+/* GCC 4.3 and above with -std=c99 or -std=gnu99 implements ISO C99
+   inline semantics, unless -fgnu89-inline is used.  */
+#if !defined __cplusplus || __GNUC_PREREQ (4,3)
+# if defined __GNUC_STDC_INLINE__ || defined __cplusplus
+#  define __extern_inline extern __inline __attribute__ ((__gnu_inline__))
+#  if __GNUC_PREREQ (4,3)
+#   define __extern_always_inline \
+  extern __always_inline __attribute__ ((__gnu_inline__, __artificial__))
+#  else
+#   define __extern_always_inline \
+  extern __always_inline __attribute__ ((__gnu_inline__))
+#  endif
+# else
+#  define __extern_inline extern __inline
+#  define __extern_always_inline extern __always_inline
+# endif
+#endif
+
+/* GCC 4.3 and above allow passing all anonymous arguments of an
+   __extern_always_inline function to some other vararg function.  */
+#if __GNUC_PREREQ (4,3)
+# define __va_arg_pack() __builtin_va_arg_pack ()
+# define __va_arg_pack_len() __builtin_va_arg_pack_len ()
+#endif
+
 /* It is possible to compile containing GCC extensions even if GCC is
    run in pedantic mode if the uses are carefully marked using the
    `__extension__' keyword.  But this is not generally available before
--- uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/nios/pt-machine.h.vanilla	2008-07-20 21:09:50.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/nios/pt-machine.h	2008-07-20 21:10:03.000000000 +0200
@@ -23,7 +23,7 @@
 #define _PT_MACHINE_H   1
 
 #ifndef PT_EI
-# define PT_EI extern inline
+# define PT_EI __extern_always_inline
 #endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/ia64/pt-machine.h.vanilla	2008-07-20 21:10:04.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/ia64/pt-machine.h	2008-07-20 21:10:10.000000000 +0200
@@ -28,7 +28,7 @@
                      size_t __child_stack_size, int __flags, void *__arg, ...);
 
 #ifndef PT_EI
-# define PT_EI extern inline __attribute__ ((always_inline))
+# define PT_EI __extern_always_inline
 #endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/cris/pt-machine.h.vanilla	2008-07-20 21:10:12.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/cris/pt-machine.h	2008-07-20 21:10:19.000000000 +0200
@@ -19,7 +19,7 @@
    Boston, MA 02111-1307, USA.  */
 
 #ifndef PT_EI
-# define PT_EI extern inline __attribute__ ((always_inline))
+# define PT_EI __extern_always_inline
 #endif
 
 PT_EI long int
--- uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/alpha/pt-machine.h.vanilla	2008-07-20 21:10:19.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/alpha/pt-machine.h	2008-07-20 21:10:23.000000000 +0200
@@ -24,7 +24,7 @@
 #define _PT_MACHINE_H   1
 
 #ifndef PT_EI
-# define PT_EI extern inline __attribute__ ((always_inline))
+# define PT_EI __extern_always_inline
 #endif
 
 #ifdef __linux__
--- uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/nios2/pt-machine.h.vanilla	2008-07-20 21:10:25.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/nios2/pt-machine.h	2008-07-20 21:10:28.000000000 +0200
@@ -22,7 +22,7 @@
 #define _PT_MACHINE_H   1
 
 #ifndef PT_EI
-# define PT_EI extern inline
+# define PT_EI __extern_always_inline
 #endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/sh/pt-machine.h.vanilla	2008-07-20 21:10:30.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/sh/pt-machine.h	2008-07-20 21:10:34.000000000 +0200
@@ -24,7 +24,7 @@
 
 #ifndef __ASSEMBLER__
 #ifndef PT_EI
-# define PT_EI extern inline __attribute__ ((always_inline))
+# define PT_EI __extern_always_inline
 #endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/m68k/pt-machine.h.vanilla	2008-07-20 21:10:35.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/m68k/pt-machine.h	2008-07-20 21:10:40.000000000 +0200
@@ -23,7 +23,7 @@
 #define _PT_MACHINE_H   1
 
 #ifndef PT_EI
-# define PT_EI extern inline __attribute__ ((always_inline))
+# define PT_EI __extern_always_inline
 #endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/mips/pt-machine.h.vanilla	2008-07-20 21:10:41.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/mips/pt-machine.h	2008-07-20 21:10:48.000000000 +0200
@@ -56,7 +56,7 @@
 
 
 #ifndef PT_EI
-# define PT_EI extern inline __attribute__ ((always_inline))
+# define PT_EI __extern_always_inline
 #endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/sparc/sparc64/pt-machine.h.vanilla	2008-07-20 21:10:50.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/sparc/sparc64/pt-machine.h	2008-07-20 21:10:54.000000000 +0200
@@ -23,7 +23,7 @@
 #define _PT_MACHINE_H   1
 
 #ifndef PT_EI
-# define PT_EI extern inline __attribute__ ((always_inline))
+# define PT_EI __extern_always_inline
 #endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/sparc/sparc32/pt-machine.h.vanilla	2008-07-20 21:10:57.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/sparc/sparc32/pt-machine.h	2008-07-20 21:11:01.000000000 +0200
@@ -23,7 +23,7 @@
 #define _PT_MACHINE_H   1
 
 #ifndef PT_EI
-# define PT_EI extern inline __attribute__ ((always_inline))
+# define PT_EI __extern_always_inline
 #endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/arm/pt-machine.h.vanilla	2008-07-20 21:11:05.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/arm/pt-machine.h	2008-07-20 21:11:17.000000000 +0200
@@ -23,7 +23,7 @@
 #define _PT_MACHINE_H   1
 
 #ifndef PT_EI
-# define PT_EI extern inline
+# define PT_EI __extern_always_inline
 #endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/powerpc/pt-machine.h.vanilla	2008-07-20 21:11:19.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/powerpc/pt-machine.h	2008-07-20 21:11:26.000000000 +0200
@@ -25,7 +25,7 @@
 #define _PT_MACHINE_H   1
 
 #ifndef PT_EI
-# define PT_EI extern inline
+# define PT_EI __extern_always_inline
 #endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/v850/pt-machine.h.vanilla	2008-07-20 21:11:28.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/v850/pt-machine.h	2008-07-20 21:11:33.000000000 +0200
@@ -12,7 +12,7 @@
  */
 
 #ifndef PT_EI
-# define PT_EI extern inline
+# define PT_EI __extern_always_inline
 #endif
 
 /* Get some notion of the current stack.  Need not be exactly the top
--- uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/x86_64/pt-machine.h.vanilla	2008-07-20 21:11:35.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/x86_64/pt-machine.h	2008-07-20 21:11:39.000000000 +0200
@@ -28,7 +28,7 @@
 
 
 # ifndef PT_EI
-#  define PT_EI extern inline __attribute__ ((always_inline))
+#  define PT_EI __extern_always_inline
 # endif
 
 extern long int testandset (int *__spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/h8300/pt-machine.h.vanilla	2008-07-20 21:11:40.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/h8300/pt-machine.h	2008-07-20 21:11:44.000000000 +0200
@@ -23,7 +23,7 @@
 #define _PT_MACHINE_H   1
 
 #ifndef PT_EI
-# define PT_EI extern inline
+# define PT_EI __extern_always_inline
 #endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/frv/pt-machine.h.vanilla	2008-07-20 21:11:44.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/frv/pt-machine.h	2008-07-20 21:11:49.000000000 +0200
@@ -25,7 +25,7 @@
 #ifndef __ASSEMBLER__
 
 #ifndef PT_EI
-# define PT_EI extern inline __attribute__ ((always_inline))
+# define PT_EI __extern_always_inline
 #endif
 
 /* Spinlock implementation; required.  */
--- uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/i386/pt-machine.h.vanilla	2008-07-20 21:07:40.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/i386/pt-machine.h	2008-07-20 21:11:54.000000000 +0200
@@ -24,7 +24,7 @@
 
 #ifndef __ASSEMBLER__
 #ifndef PT_EI
-# define PT_EI extern inline __attribute__ ((always_inline))
+# define PT_EI __extern_always_inline
 #endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/bfin/pt-machine.h.vanilla	2008-07-20 21:11:55.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/bfin/pt-machine.h	2008-07-20 21:11:59.000000000 +0200
@@ -22,7 +22,7 @@
 #define _PT_MACHINE_H   1
 
 #ifndef PT_EI
-# define PT_EI extern inline
+# define PT_EI __extern_always_inline
 #endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/sh64/pt-machine.h.vanilla	2008-07-20 21:12:00.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads.old/sysdeps/sh64/pt-machine.h	2008-07-20 21:12:04.000000000 +0200
@@ -23,7 +23,7 @@
    Boston, MA 02111-1307, USA.  */
 
 #ifndef PT_EI
-# define PT_EI extern inline
+# define PT_EI __extern_always_inline
 #endif
 
 /* Spinlock implementation; required.  */
--- uClibc-0.9.29/libpthread/linuxthreads/sysdeps/s390/s390-32/pt-machine.h.vanilla	2008-07-20 21:12:04.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads/sysdeps/s390/s390-32/pt-machine.h	2008-07-20 21:12:09.000000000 +0200
@@ -23,7 +23,7 @@
 #define _PT_MACHINE_H   1
 
 #ifndef PT_EI
-# define PT_EI extern inline __attribute__ ((always_inline))
+# define PT_EI __extern_always_inline
 #endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads/sysdeps/s390/s390-64/pt-machine.h.vanilla	2008-07-20 21:12:09.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads/sysdeps/s390/s390-64/pt-machine.h	2008-07-20 21:12:13.000000000 +0200
@@ -23,7 +23,7 @@
 #define _PT_MACHINE_H   1
 
 #ifndef PT_EI
-# define PT_EI extern inline __attribute__ ((always_inline))
+# define PT_EI __extern_always_inline
 #endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads/sysdeps/ia64/pt-machine.h.vanilla	2008-07-20 21:12:13.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads/sysdeps/ia64/pt-machine.h	2008-07-20 21:12:17.000000000 +0200
@@ -24,7 +24,7 @@
 #include <ia64intrin.h>
 
 #ifndef PT_EI
-# define PT_EI extern inline __attribute__ ((always_inline))
+# define PT_EI __extern_always_inline
 #endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads/sysdeps/cris/pt-machine.h.vanilla	2008-07-20 21:12:17.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads/sysdeps/cris/pt-machine.h	2008-07-20 21:12:21.000000000 +0200
@@ -22,7 +22,7 @@
 #define _PT_MACHINE_H   1
 
 #ifndef PT_EI
-# define PT_EI extern inline __attribute__ ((always_inline))
+# define PT_EI __extern_always_inline
 #endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads/sysdeps/alpha/pt-machine.h.vanilla	2008-07-20 21:12:21.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads/sysdeps/alpha/pt-machine.h	2008-07-20 21:12:26.000000000 +0200
@@ -24,7 +24,7 @@
 #define _PT_MACHINE_H   1
 
 #ifndef PT_EI
-# define PT_EI extern inline __attribute__ ((always_inline))
+# define PT_EI __extern_always_inline
 #endif
 
 #ifdef __linux__
--- uClibc-0.9.29/libpthread/linuxthreads/sysdeps/sh/pt-machine.h.vanilla	2008-07-20 21:12:26.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads/sysdeps/sh/pt-machine.h	2008-07-20 21:12:31.000000000 +0200
@@ -24,7 +24,7 @@
 
 #ifndef __ASSEMBLER__
 #ifndef PT_EI
-# define PT_EI extern inline __attribute__ ((always_inline))
+# define PT_EI __extern_always_inline
 #endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads/sysdeps/m68k/pt-machine.h.vanilla	2008-07-20 21:12:33.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads/sysdeps/m68k/pt-machine.h	2008-07-20 21:12:37.000000000 +0200
@@ -23,7 +23,7 @@
 #define _PT_MACHINE_H   1
 
 #ifndef PT_EI
-# define PT_EI extern inline __attribute__ ((always_inline))
+# define PT_EI __extern_always_inline
 #endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads/sysdeps/mips/pt-machine.h.vanilla	2008-07-20 21:12:40.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads/sysdeps/mips/pt-machine.h	2008-07-20 21:12:44.000000000 +0200
@@ -28,7 +28,7 @@
 #include <sys/tas.h>
 
 #ifndef PT_EI
-# define PT_EI extern inline __attribute__ ((always_inline))
+# define PT_EI __extern_always_inline
 #endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads/sysdeps/hppa/pt-machine.h.vanilla	2008-07-20 21:12:45.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads/sysdeps/hppa/pt-machine.h	2008-07-20 21:12:48.000000000 +0200
@@ -25,7 +25,7 @@
 #include <bits/initspin.h>
 
 #ifndef PT_EI
-# define PT_EI extern inline __attribute__ ((always_inline))
+# define PT_EI __extern_always_inline
 #endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads/sysdeps/sparc/sparc64/pt-machine.h.vanilla	2008-07-20 21:12:49.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads/sysdeps/sparc/sparc64/pt-machine.h	2008-07-20 21:12:52.000000000 +0200
@@ -23,7 +23,7 @@
 #define _PT_MACHINE_H   1
 
 #ifndef PT_EI
-# define PT_EI extern inline __attribute__ ((always_inline))
+# define PT_EI __extern_always_inline
 #endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads/sysdeps/sparc/sparc32/pt-machine.h.vanilla	2008-07-20 21:12:53.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads/sysdeps/sparc/sparc32/pt-machine.h	2008-07-20 21:12:56.000000000 +0200
@@ -23,7 +23,7 @@
 #define _PT_MACHINE_H   1
 
 #ifndef PT_EI
-# define PT_EI extern inline __attribute__ ((always_inline))
+# define PT_EI __extern_always_inline
 #endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads/sysdeps/arm/pt-machine.h.vanilla	2008-07-20 21:12:58.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads/sysdeps/arm/pt-machine.h	2008-07-20 21:13:02.000000000 +0200
@@ -23,7 +23,7 @@
 #define _PT_MACHINE_H   1
 
 #ifndef PT_EI
-# define PT_EI extern inline __attribute__ ((always_inline))
+# define PT_EI __extern_always_inline
 #endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads/sysdeps/powerpc/powerpc32/pt-machine.h.vanilla	2008-07-20 21:13:02.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads/sysdeps/powerpc/powerpc32/pt-machine.h	2008-07-20 21:13:08.000000000 +0200
@@ -26,7 +26,7 @@
 #define _PT_MACHINE_H   1
 
 #ifndef PT_EI
-# define PT_EI extern inline __attribute__ ((always_inline))
+# define PT_EI __extern_always_inline
 #endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads/sysdeps/powerpc/powerpc64/pt-machine.h.vanilla	2008-07-20 21:13:10.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads/sysdeps/powerpc/powerpc64/pt-machine.h	2008-07-20 21:13:15.000000000 +0200
@@ -25,7 +25,7 @@
 #define _PT_MACHINE_H   1
 
 #ifndef PT_EI
-# define PT_EI extern inline __attribute__ ((always_inline))
+# define PT_EI __extern_always_inline
 #endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads/sysdeps/x86_64/pt-machine.h.vanilla	2008-07-20 21:13:15.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads/sysdeps/x86_64/pt-machine.h	2008-07-20 21:13:22.000000000 +0200
@@ -28,7 +28,7 @@
 
 
 # ifndef PT_EI
-#  define PT_EI extern inline __attribute__ ((always_inline))
+#  define PT_EI __extern_always_inline
 # endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads/sysdeps/i386/pt-machine.h.vanilla	2008-07-20 21:13:22.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads/sysdeps/i386/pt-machine.h	2008-07-20 21:13:27.000000000 +0200
@@ -29,7 +29,7 @@
 
 #ifndef __ASSEMBLER__
 #ifndef PT_EI
-# define PT_EI extern inline __attribute__ ((always_inline))
+# define PT_EI __extern_always_inline
 #endif
 
 extern long int testandset (int *spinlock);
--- uClibc-0.9.29/libpthread/linuxthreads/sysdeps/i386/i686/pt-machine.h.vanilla	2008-07-20 21:13:28.000000000 +0200
+++ uClibc-0.9.29/libpthread/linuxthreads/sysdeps/i386/i686/pt-machine.h	2008-07-20 21:13:32.000000000 +0200
@@ -23,7 +23,7 @@
 #define _PT_MACHINE_H	1
 
 #ifndef PT_EI
-# define PT_EI extern inline __attribute__ ((always_inline))
+# define PT_EI __extern_always_inline
 #endif
 #include "kernel-features.h"
 
